modules/objdetect/src/qrcode.cpp

```diff
@@ -70,151 +67,153 @@ void QRDetect::init(const Mat& src, double eps_vertical_, double eps_horizontal_
-                pixels_position[index] = x;
-                index++;
-            }
-        }
-        pixels_position[index] = width_bin_barcode - 1;
-        index++;
-        for (size_t i = 2; i < index - 4; i+=2)
+        const int width_m_img  = m_img.cols;
+        const size_t test_lines_size = 5;
+        double test_lines[test_lines_size];
+        const size_t count_pixels_position = 1024;
+        size_t pixels_position[count_pixels_position];
+        size_t index = 0;
+        vector<Point2f> local_result;
+
+        for (int y = range.start; y < range.end; y++)
         {
-            test_lines[0] = static_cast<double>(pixels_position[i - 1] - pixels_position[i - 2]);
-            test_lines[1] = static_cast<double>(pixels_position[i    ] - pixels_position[i - 1]);
-            test_lines[2] = static_cast<double>(pixels_position[i + 1] - pixels_position[i    ]);
-            test_lines[3] = static_cast<double>(pixels_position[i + 2] - pixels_position[i + 1]);
-            test_lines[4] = static_cast<double>(pixels_position[i + 3] - pixels_position[i + 2]);
-
-            double length = 0.0, weight = 0.0;
-
-            for (size_t j = 0; j < test_lines_size; j++) { length += test_lines[j]; }
-
-            if (length == 0) { continue; }
-            for (size_t j = 0; j < test_lines_size; j++)
+            int pos = 0;
+            for (; (pos < width_m_img) && (m_img.ptr<uint8_t>(y)[pos] == 0); pos++) {}
+            if (pos < width_m_img)
             {
-                if (j == 2) { weight += fabs((test_lines[j] / length) - 3.0/7.0); }
-                else        { weight += fabs((test_lines[j] / length) - 1.0/7.0); }
-            }
+                index = 0;
+                pixels_position[index] = pixels_position[index + 1] = pixels_position[index + 2] = pos;
+                index +=3;
 
-            if (weight < eps_vertical)
-            {
-                Vec3d line;
-                line[0] = static_cast<double>(pixels_position[i - 2]);
-                line[1] = y;
-                line[2] = length;
-                result.push_back(line);
-            }
-        }
-    }
-    return result;
-}
-
-vector<Point2f> QRDetect::separateVerticalLines(const vector<Vec3d> &list_lines)
-{
-    vector<Vec3d> result;
-    int temp_length = 0;
-    uint8_t next_pixel;
-    vector<double> test_lines;
-
-
-    for (size_t pnt = 0; pnt < list_lines.size(); pnt++)
-    {
-        const int x = cvRound(list_lines[pnt][0] + list_lines[pnt][2] * 0.5);
-        const int y = cvRound(list_lines[pnt][1]);
-
-        // --------------- Search vertical up-lines --------------- //
+                uint8_t future_pixel = 255;
+                for (int x = pos; x < width_m_img; x++)
+                {
+                    if (m_img.ptr<uint8_t>(y)[x] == future_pixel)
+                    {
+                        future_pixel = 255 - future_pixel;
+                        pixels_position[index] = x;
+                        index++;
```

Comment:
Still have the problem with uncontrolled buffer overflow. `count_pixels_position` is useless.

Use `cv::AutoBuffer<>()`/`std::vector<>()` for dynamic allocations.

BTW, What is performance on 4K 1-pixel chessboard-like image?